clinvar_ingest/cloud/gcs.py

```diff
@@ -17,11 +17,50 @@ def parse_blob_uri(uri: str) -> storage.Blob:
     )
 
 
-def copy_file_to_bucket(local_file_uri: str, remote_blob_uri):
+def copy_file_to_bucket(
+    local_file_uri: str, remote_blob_uri: str, client=storage.Client()
+):
     """
     Upload the contents of file `local_file_uri` on local filesystem, to `remote_blob_uri` in
     """
     _logger.info(f"Uploading {local_file_uri} to {remote_blob_uri}")
     blob = parse_blob_uri(remote_blob_uri)
     blob.upload_from_filename(client=client, filename=local_file_uri)
     _logger.info(f"Finished uploading {local_file_uri} to {remote_blob_uri}")
+
+
+def blob_writer(blob_uri: str, client=storage.Client(), binary=True) -> storage.Blob:
+    """
+    Returns a file-like object that can be used to write to the blob at `blob_uri`
+    """
+    blob = parse_blob_uri(blob_uri, client=client)
+    return blob.open("wb" if binary else "w")
+
+
+def http_upload_shell(http_uri: str, blob_uri: str):
+    """
+    Upload the contents of `http_uri` to `blob_uri` using curl and gsutil
+    """
+    p = subprocess.run(
+        ["bash", "-c", f"curl {http_uri} | gcloud storage cp - {blob_uri}"],
+        capture_output=True,
+        check=True,
+    )
+
+    if p.returncode != 0:
+        raise RuntimeError(f"curl failed:\n{p.stdout}\n{p.stderr}")
+
+
+def http_upload_urllib(
+    http_uri: str, blob_uri: str, client=storage.Client(), chunk_size=1024 * 16
+):
+    _logger.info(f"Uploading {http_uri} to {blob_uri}")
+    with urllib.request.urlopen(http_uri) as f:
```

Comment:
I just tried `urllib.requests.urlopen`, which I think uses ftplib. At least on my home network it seemed like the http method was downloading faster than the ftp url through `urlopen`. Might be specific to python because I'd expect ftp protocol to be faster in general than http.

I can do a quick test on a VM to see